src/v/cluster/log_eviction_stm.cc

```diff
@@ -333,8 +356,7 @@ ss::future<> log_eviction_stm::apply(model::record_batch batch) {
 
         /// Set the new in memory start offset
         _delete_records_eviction_offset = truncate_offset;
-        /// Wake up the background reaping thread
-        _reap_condition.signal();
+        co_await enqueue_eviction_event(truncate_offset, wait_for_success::yes);
```

Comment:
nit: not sure if there's a better name, but at first glance it's a unintuitive that `wait_for_success::yes` doesn't wait for the event to be processed. Maybe `retry_event_until_success` would clarify? I don't feel strongly about it